src/components/Forecast/Widget.jsx

```diff
@@ -0,0 +1,181 @@
+
+    if (city && country) {
+      fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city},${country}&appid=${myAPIKey}&units=metric`)
+        .then((e) => e.json())
+        .then((data) => {
+          if (data.cod == '404') {
+            this.setState({
+              isClicked: true,
+              cityFound: false,
+              city: undefined,
+              country: undefined,
+              temperature: undefined,
+              humidity: undefined,
+              description: undefined,
+              wind: undefined,
+              pressure: undefined,
+              sunrise: undefined,
+              sunrise: undefined,
+              icon: undefined,
+              in3hours: undefined,
+              in6hours: undefined,
+              in9hours: undefined,
+              in12hours: undefined,
+              day1: undefined,
+              day2: undefined,
+              day3: undefined,
+              day4: undefined,
+              day5: undefined,
+              error: data.message,
+            })
+          } else {
+            this.setState({
+              city: data.name,
+              country: data.sys.country,
+              temperature: data.main.temp,
+              humidity: data.main.humidity,
+              description: data.weather[0].description,
+              icon: data.weather[0].icon,
+              wind: data.wind,
+              pressure: data.main.pressure,
+              sunrise: data.sys.sunrise,
+              sunset: data.sys.sunset,
+              error: '',
+            })
+            fetch(`http://api.openweathermap.org/data/2.5/forecast?q=${city},${country}&appid=${myAPIKey}&units=metric`)
+              .then((e) => e.json())
+              .then((data) => {
+                this.setState({
+                  cityFound: true,
+                  isClicked: true,
+                  in3hours: data.list[0],
+                  in6hours: data.list[1],
+                  in9hours: data.list[2],
+                  in12hours: data.list[3],
+                  day1: data.list[7],
+                  day2: data.list[15],
+                  day3: data.list[23],
+                  day4: data.list[31],
+                  day5: data.list[39],
+                })
+              })
+          }
+        })
+    } else {
+      this.setState({
+        isClicked: true,
+        city: undefined,
+        country: undefined,
+        temperature: undefined,
+        humidity: undefined,
+        description: undefined,
+        wind: undefined,
+        pressure: undefined,
+        sunrise: undefined,
+        sunrise: undefined,
+        icon: undefined,
+        in3hours: undefined,
+        in6hours: undefined,
+        in9hours: undefined,
+        in12hours: undefined,
+        day1: undefined,
+        day2: undefined,
+        day3: undefined,
+        day4: undefined,
+        day5: undefined,
+        error: 'Please enter the city and country code.',
+      })
+    }
+  }
```

Comment:
It's really hard to see what's going on here! What I'd suggest for this would be to save the empty state object in a variable, so you don't have to copy/paste like this.